### PR TITLE
Add tutorial on using Auction Slot to save on AMM fees

### PR DESCRIPTION
**Pull request by mDuo13**
Still needs a fair bit of cleaning up but I think this tutorial is now feature-complete.

The new tutorial is at /docs/tutorials/javascript/trade-on-ledger/use-amm-auction-slot-for-lower-fees

- [x] Update sidebar and add landing page for new Javascript/Trade in the DEX category
- [x] Finish moving out sample code for calculating auction price for existing LPs
- [x] Convert "Tip"/"Note"/etc to proper (Redocly-formatted) callouts

### CODE REVIEW

File: docs/tutorials/javascript/trade-on-ledger/use-amm-auction-slot-for-lower-fees.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
+# Use the AMM Auction Slot for Lower Fees
+
+When trading on the XRP Ledger's decentralized exchange (DEX), you can sometimes save on trading fees by buying the auction slot built into an [Automated Market Maker (AMM)](../../../concepts/tokens/decentralized-exchange/automated-market-makers.md). This tutorial shows how to identify when the auction slot can save you money and what transactions to send to use it, with example code in JavaScript.
+
+For a simpler example of trading currencies in the XRP Ledger's DEX, see [Trade in the Decentralized Exchange](../../how-tos/use-tokens/trade-in-the-decentralized-exchange.md).
+
+This tutorial does not exhaustively cover all possible market conditions and circumstances. Always exercise caution and trade at your own risk.
```
Comment (amarantha-k) [inline docs/tutorials/javascript/trade-on-ledger/use-amm-auction-slot-for-lower-fees.md:8]:
```suggestion
{% admonition type="warning" name="CAUTION" %}
This tutorial does not exhaustively cover all possible market conditions and circumstances. Always exercise caution and trade at your own risk.
{% /admonition %}
```

File: docs/tutorials/javascript/trade-on-ledger/use-amm-auction-slot-for-lower-fees.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+## Prerequisites
+
+- You need a connection to the XRP Ledger network. As shown in this tutorial, you can use public servers for testing.
+- You should be familiar with basic usage of the [xrpl.js client library](https://github.com/XRPLF/xrpl.js/). See [Get Started Using JavaScript](../build-apps/get-started.md) for setup steps.
+- The AMM for the asset pair you want to trade must already exist in the ledger. This tutorial uses an AMM on Testnet which has been set up in advance. For instructions on creating an AMM for a different currency pair, see [Create an Automated Market Maker](../../how-tos/use-tokens/create-an-automated-market-maker.md).
```
Comment (amarantha-k) [inline docs/tutorials/javascript/trade-on-ledger/use-amm-auction-slot-for-lower-fees.md]: Note the asset pair being used in this tutorial (XRP <> TST)?

File: docs/tutorials/javascript/trade-on-ledger/use-amm-auction-slot-for-lower-fees.md [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+
+## Background
```
Comment (oeggert) [inline docs/tutorials/javascript/trade-on-ledger/use-amm-auction-slot-for-lower-fees.md]: Piggybacking off Amarantha's comment above, maybe just remove this section and add the info to the last bullet point in Prereqs?

File: docs/tutorials/javascript/trade-on-ledger/use-amm-auction-slot-for-lower-fees.md [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+This tutorial uses an AMM instance that has been set up on the XRP Ledger Testnet in advance, connecting the following assets:
+
+| Currency Code | Issuer | Notes |
+|---|---|---|
+| XRP | N/A | Testnet XRP is functionally similar to XRP, but holds no real-world value. You can get it for free from a [faucet](/resources/dev-tools/xrp-faucets).
+| TST | `rP9jPyP5kyvFRb6ZiRghAGw5u8SGAmU4bd` | A test token pegged to XRP at a rate of approximately 10 XRP per 1 TST. The issuer has existing Offers on the XRP Ledger Testnet to buy and sell these tokens. This token has no [transfer fee](../../../concepts/tokens/transfer-fees.md) or [Tick Size](../../../concepts/tokens/decentralized-exchange/ticksize.md) set. |
+
+
+## Steps
```
Comment (oeggert) [inline docs/tutorials/javascript/trade-on-ledger/use-amm-auction-slot-for-lower-fees.md:29]: This feels like a deviation from how we normally format our tutorials. Should we treat the "Steps" section as an overview of steps? Our other tutorials treat them as explicit instructions.

File: docs/tutorials/javascript/trade-on-ledger/use-amm-auction-slot-for-lower-fees.md [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+
+{% code-snippet file="/_code-samples/auction-slot/js/auction-slot.js" language="js" from="// Actually bid" before="// Trade using the discount" /%}
+
+{% admonition type="success" name="Tip" %}The amounts that LP Tokens get rounded can be surprising. If you don't plan on holding LP Tokens after bidding, you should set `BidMin` to the same as `BidMax` so that you aren't left with a trust line that contains a very tiny amount of LP Tokens that weren't spent on the auction price, and you don't have to meet the XRP reserve for that trust line.{% /admonition %}
+
+
+### 8. Trade using the discount
+
+If your previous transaction was successful, you should now be the auction slot holder until you are outbid or 24 hours have passed, whichever comes first. You can immediately use this opportunity to make the XRP for TST trade that you wanted to make in the first place. There are several ways to make trades in the XRP Ledger, but sending an OfferCreate transaction is the most conventional, as in the following code:
```
Comment (oeggert) [inline docs/tutorials/javascript/trade-on-ledger/use-amm-auction-slot-for-lower-fees.md:148]: Is there any way to check if you're the auction slot holder? Like will you get any kind of notification if someone outbids you and takes the spot?

File: docs/tutorials/javascript/trade-on-ledger/use-amm-auction-slot-for-lower-fees.md [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+
+The following functions implement `feeMult`, `feeMultHalf`, and `feeDecimal` in JavaScript:
+
+{% code-snippet file="/_code-samples/auction-slot/js/amm-formulas.js" language="js" from="/* Convert a trading fee to a value that" before="/* Implement the AMM SwapOut formula" /%}
+
+The AssetIn and SwapOut functions depend on these helper functions.
+
+### solveQuadraticEq
+
+This helper function is solves the quadratic equation, which you may remember from high-school mathematics.
```
Comment (oeggert) [inline docs/tutorials/javascript/trade-on-ledger/use-amm-auction-slot-for-lower-fees.md]:
```suggestion
This helper function solves the quadratic equation, which you may remember from high-school mathematics.
```
Maybe remove the comment about high-school mathematics? How conversational are we trying to be?

**Review by oeggert — APPROVED** [review]
Approving this PR since the code and tutorial is fine, but consider suggestions.

File: docs/tutorials/javascript/trade-on-ledger/use-amm-auction-slot-for-lower-fees.md (hunk 4 above)
Comment (mDuo13) [reply docs/tutorials/javascript/trade-on-ledger/use-amm-auction-slot-for-lower-fees.md:29]: If you notice the hierarchy of the headers, the "Steps" section does in fact include the full actual steps; the section introduction here provides a high-level overview of the steps to aid understanding.

To make it a little easier to see the connection, I moved the source code link up out of the "Steps" section.

Comment (mDuo13) [reply docs/tutorials/javascript/trade-on-ledger/use-amm-auction-slot-for-lower-fees.md:29]: It's not ideal that both the overview and the actual process have step numbers and they don't match. I'm going to try turning it into a bulleted list with bolded introductions whose numbers match the actual steps.

File: docs/tutorials/javascript/trade-on-ledger/use-amm-auction-slot-for-lower-fees.md (hunk 5 above)
Comment (mDuo13) [reply docs/tutorials/javascript/trade-on-ledger/use-amm-auction-slot-for-lower-fees.md:148]: You can query the AMM status at any time to find out who holds the slot, but I don't think there's a way to get a subscription/notification that you were outbid.

**Pull request merged**